src/interfaces/wallet.cpp

```diff
@@ -481,16 +484,24 @@ class WalletClientImpl : public ChainClient
 {
 public:
     WalletClientImpl(Chain& chain, std::vector<std::string> wallet_filenames)
-        : m_chain(chain), m_wallet_filenames(std::move(wallet_filenames))
+        : m_wallet_filenames(std::move(wallet_filenames))
     {
+        m_context.chain = &chain;
     }
     void registerRpcs() override
     {
-        g_rpc_chain = &m_chain;
-        return RegisterWalletRPCCommands(m_chain, m_rpc_handlers);
+        for (const CRPCCommand& command : GetWalletRPCCommands()) {
+            m_rpc_commands.emplace_back(command.category, command.name, [this, &command](const JSONRPCRequest& request, UniValue& result, bool last_handler) {
+                util::Ref context{m_context};
+                JSONRPCRequest wallet_request{context};
+                wallet_request = request;
+                return command.actor(wallet_request, result, last_handler);
```

Comment:
Instead of adding the `operator=` (which doesn't copy m_context) maybe overload constructor so that here would be `return command.actor({request, m_context}, result, last_handler)`. I've pushed 5e433dcb6fc3106281d14330879acf0ef1160b62 with this suggestion.